File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(0);
        ListNode tail = dummy;

        ListNode ls1 = list1;
        ListNode ls2 = list2;
        
        while (ls1!=null && ls2!=null) {
            if (ls1.val<ls2.val) {
                tail.next=ls1;
                ls1=ls1.next;
            } else {
                tail.next=ls2;
                ls2=ls2.next;
            }
            tail=tail.next;
        }

        if (ls1!=null) {
            tail.next=ls1;
        }

        else {
            tail.next=ls2;
        }

        return dummy.next;
    }
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {

        ListNode* first = head;

        int size = 0;
        
        while (first != nullptr)
        {
            first = first->next;
            size++;
        }

        if (size == n)
        {
            return head->next;
        }

        ListNode* second = head;

        for (int i = 0; i < size - n - 1; i++)
        {

            second = second->next;

        }

        second->next = second->next->next;

        return head;
    }

};